Core/Business/Addon/Theme/Theme.php

```diff
@@ -25,8 +25,20 @@
  */
 namespace PrestaShop\PrestaShop\Core\Business\Addon\Theme;
 
-abstract class AbstractTheme implements AddonInterface
+use PrestaShop\PrestaShop\Core\Business\Addon\AddonInterface;
+
+class Theme implements AddonInterface
 {
+    public function __construct($directory)
+    {
+        $this->directory = rtrim($directory, '/') . '/';
+        $this->setPropertiesFromJson();
```

Comment:
Reading from the filesystem should not be done by the constructor, as it prevents me from instantiating a "Theme" without passing a valid directory (yes we do it all the time, but not in the `Core` subdirectory :p). It's going to make testing the class hard to test.